docs/docs/reference-dom-elements.md

```diff
@@ -1,22 +1,91 @@
 ---
-id: tags-and-attributes
-title: Tags and Attributes
-permalink: docs-old/tags-and-attributes.html
-prev: component-specs.html
-next: events.html
+id: dom-elements
+title: DOM Elements
+layout: docs
+category: Reference
+permalink: docs/dom-elements.html
 ---
 
-## Supported Elements
+React implements a browser-independent DOM system for performance and cross-browser compatibility. We took the opportunity to clean up a few rough edges in browser DOM implementations.
 
-React attempts to support all common elements in both HTML and SVG. Any lower case tag in JSX will be rendered to an element with that tag. SVG elements must be contained within an `<svg>` element to work properly.
+In React, all DOM properties and attributes (including event handlers) should be camelCased. For example, the HTML attribute `tabindex` corresponds to the attribute `tabIndex` in React. The exception is `aria-*` and `data-*` attributes, which should be lowercased.
 
-### Using `React.DOM` Factory methods
+## Differences In Attributes
 
-If you aren't using JSX and are using the `React.DOM.*` API to create elements, then you are slightly more limited and there is list of supported elements that will be available on that API.
+There are a number of attributes that work differently between React and HTML:
 
-#### HTML Elements
+### checked
 
-The following HTML elements are supported in `React.DOM.*`:
+The `checked` attribute is supported by `<input>` components of type `checkbox` or `radio`. You can use it to set whether the component is checked. This is useful for building controlled components.
+
+### className
+
+Since `class` is a reserved word in JavaScript, React elements use `className` instead.
+
+### dangerouslySetInnerHTML
+
+`dangerouslySetInnerHTML` is React's replacement for using `innerHTML` in the browser DOM. In general, setting HTML from code is risky because it's easy to inadvertently expose your users to a [cross-site scripting (XSS)](https://en.wikipedia.org/wiki/Cross-site_scripting) attack. So, you can set HTML directly from React, but you have to type out `dangerouslySetInnerHTML` to remind yourself that it's dangerous. For example:
+
+```js
+function createMarkup() {
+  return {__html: 'First &middot; Second'};
+}
+
+function MyComponent() {
+  return <div dangerouslySetInnerHTML={createMarkup()} />;
+}
+```
+
+### htmlFor
+
+Since `for` is a reserved word in JavaScript, React elements use `htmlFor` instead.
+
+### key
+
+`key` in React doesn't correspond to anything in HTML. It's an optional unique identifier for components in lists. Assigning it a key that persists makes sure the component stays. You can read more [here](/react/docs/reconciliation.html).
```

Comment:
It's not quite optional. React loudly complains if it is missing.

Giving it a key just means component gets reordered efficiently. It does not determine whether it "stays".